.github/workflows/main.yml

```diff
@@ -0,0 +1,73 @@
+name: CI
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  js_build_and_test:
+    name: JS Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '10'
+      - run: npm ci
+      - run: npm run eslint
+      - run: npm run test
+  php_build_and_test:
+    name: PHP Build and Test
+    runs-on: ubuntu-latest
+    services:
+      mariadb:
+        image: mariadb:latest
+        ports:
+          - 3306:3306
+        env:
+          MYSQL_USER: wp_test
+          MYSQL_PASSWORD: wp_test
+          MYSQL_DATABASE: wordpress_default
+          MYSQL_ROOT_PASSWORD: root
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    steps:
+      - name: Checkout latest WCS
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Switch to PHP 7.4
+        run: |
+          sudo update-alternatives --set php /usr/bin/php7.4
+          sudo update-alternatives --set phar /usr/bin/phar7.4
+          sudo update-alternatives --set phar.phar /usr/bin/phar.phar7.4
+      - name: Verify PHP version
+        run: |
+          lsb_release -a
+          sudo update-alternatives --config php
+          sudo update-alternatives --config phar
+          sudo update-alternatives --config phar.phar
+          php -v
+      - name: Verify MySQL connection
+        run: |
+          mysql --version
+          sudo apt-get install -y mysql-client
+          mysql --host 127.0.0.1 --port ${{ job.services.mariadb.ports['3306'] }} -uroot -proot -e "SHOW DATABASES"
+      - name: Install composer
+        run: |
+          php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+          php -r "if (hash_file('sha384', 'composer-setup.php') === '756890a4488ce9024fc62c56153228907f1545c228516cbf63f885e036d37e9a59d27d63f46af1d4d07ee0f76181c7d3') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
+          php composer-setup.php
+          php -r "unlink('composer-setup.php');"
+          sudo mv composer.phar /usr/local/bin/composer
+          composer install
+      - name: Setup WP and WC
+        run: tests/bin/install-wc-tests.sh wordpress_test root root 127.0.0.1 5.6.2 release/5.0
```

Comment:
Using the same setup script we use for Travis. It would be nice if we can make this into our own WooCommerce Action.